pkg/source/source.go

```diff
@@ -248,16 +293,16 @@ func (is *Informer) String() string {
 	return fmt.Sprintf("informer source: %p", is.Informer)
 }
 
-var _ Source = Func(nil)
+var _ Source = Func[reconcile.Request](nil)
 
 // Func is a function that implements Source.
-type Func func(context.Context, workqueue.RateLimitingInterface) error
+type Func[request comparable] func(context.Context, workqueue.TypedRateLimitingInterface[request]) error
```

Comment:
This is a breaking change but I think `source.Func` is barely used. Prior to the first generics PR, I didn't know it exists.

Reply:
Yeah, updated it to do that. I initially thought its pretty much completely unused, but turns out there are a few users: https://github.com/search?type=code&q=source.Func%28+language%3AGo